### PR TITLE
Chore/dev 1047 improved typings

### PR DESCRIPTION
**Pull request by kulgan**

### CODE REVIEW

File: indexclient/client.pyi [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+]
+
+class URLMetadataTuple(NamedTuple):
+    url: str
+    type: IndexFileLocationType
+    state: IndexFileState
+
+class UrlMetadataDict(TypedDict):
+    state: IndexFileState
+    type: IndexFileLocationType
```
Comment (ProfOak) [inline indexclient/client.pyi:47]: Does this allow for other dictionary keys? For example we use `md5` as a key for backed up urls in the runners.

Comment (kulgan) [reply indexclient/client.pyi:47]: yea, typeddicts are essentially dictionaries with known keys

File: dev-requirements.in [hunk 2]
```diff
@@ -1,16 +1,17 @@
--c requirements.txt
```
Comment (qiaouchicago) [inline dev-requirements.in:1]: We do not need this to be compatible with requirements.txt?

**Comment by kulgan** [comment]
no longer needed

**Pull request closed without merging**